### PR TITLE
test-case: add keyword detection test support

### PR DESCRIPTION
**Pull request by keqiaozhang**
add one new test case to check the keyword detection and use
the wavetool to do the analysis.

Signed-off-by: Zhang Keqiao <keqiao.zhang@linux.intel.com>

### CODE REVIEW

File: test-case/check-keyword-detection.sh [hunk 1]
```diff
@@ -0,0 +1,159 @@
+#!/bin/bash
+
+set -e
+
+##
+## Case Name: check_keyword_detection
```
Comment (aiChaoSONG) [inline test-case/check-keyword-detection.sh:6]: make this the same as file name: check-keyword-detection

File: test-case/check-keyword-detection.sh [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+}
+
+for i in $(seq 1 "$loop_cnt")
+do
+    for fmt in $fmts
+    do
+	if [ "$fmt" == "S24_LE" ]; then
+        dlogi "S24_LE is not supported, skip to test this format"
+        continue
+    fi
```
Comment (aiChaoSONG) [inline test-case/check-keyword-detection.sh:129]: indent error in this if code block

File: test-case/check-keyword-detection.sh [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    _store_default_wov_config_bolb || die "Failed to dump KWD config blob "
+    _get_default_pt_hd || die "Failed to get the default preamble_time and history_depth"
+    _update_blob || die "Failed to update the KWD config blob"
+    # generate the wav files for test
+    wavetool.py -g wov -A 0.25 1. -D 4. 6. -B "${fmt%_*}" -o "$test_file"
+
+    # play the testing wav first
+    dlogc "aplay -Dplughw:1,0 $test_file"
+    aplay -Dplughw:1,0 "$test_file" & aplayPID=$!
+	dlogi "Testing: iteration $i with $fmt format\n"
```
Comment (aiChaoSONG) [inline test-case/check-keyword-detection.sh:141]: indent error  in this line. Please delete the `\n` at the end of the line.

File: test-case/check-keyword-detection.sh (hunk 1 above)
Comment (keqiaozhang) [reply test-case/check-keyword-detection.sh]: Fixed.

File: test-case/check-keyword-detection.sh (hunk 2 above)
Comment (keqiaozhang) [reply test-case/check-keyword-detection.sh]: Fixed.

File: test-case/check-keyword-detection.sh (hunk 3 above)
Comment (keqiaozhang) [reply test-case/check-keyword-detection.sh]: Fixed.

File: test-case/check-keyword-detection.sh [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+# adjust wov PGA volume
+amixer cset name="$wov_kctl_pga" 45
+
+# store the default config blob
+_store_default_wov_config_bolb()
+{
+    if [ ! -f "$def_blob" ]; then
+        sof-ctl -Dhw:0 -n "$wov_kctl_id" -r -o "$def_blob"
+	    dlogi "default blob:"
```
Comment (fredoh9) [inline test-case/check-keyword-detection.sh:90]: two lines have TAB

**Comment by marc-hb** [comment]
How did this test work without OFS fix #350 ? How was this test tested?

All but one shellcheck warning are fixed by trivial two-lines commit #352

**Pull request merged**